jax/experimental/compilation_cache/file_system_cache.py

```diff
@@ -0,0 +1,48 @@
+# Copyright 2021 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+## Core functionality:
+##create my own directory in experimental folder in jax
+##file for this class then seperate file for tests
+
+
+from typing import Optional
+import os
+import tempfile
+
+class FileSystemCache:
+
+  def __init__(self, path: str):
+    """Sets up a cache at 'path'. Cached values may already be present."""
+    if not os.path.isdir(path):
+      os.mkdir(path)
+    self._path  = path
+
+  def get(self, key: str) -> Optional[bytes]:
+    """Returns None if 'key' isn't present."""
+    if(key==""):
+      raise ValueError("key cannot be empty, you cannot create a directory path for a new file from the home directory which has already been created.")
```

Comment:
I don't really understand the part about creating a new file from the home directory. Let's just leave it at "key cannot be empty".